[18.0][MIG] base_technical_features: Migration to 18.0

Kimkhoi3010 opened the pull request:
I found only 1 commit to port but it was ignored, I added it to blacklist:: [[17.0][FW] base_technical_features: port from 14.0](https://github.com/Kimkhoi3010/server-ux/pull/3)

# Reason for change:
The original logic in _postprocess_access_rights removes base.group_no_one for non-admin users, which restricts their access to certain UI components. Extending _get_view ensures that technical users (part of base_technical_features.group_technical_features) still have access to these components without needing full admin privileges.

StefanRijnhart committed:
[ADD] Access to technical features as a user preference

StefanRijnhart committed:
[MIG] base_technical_features -> 10.0 (#568)

Using new base model inheritance

StefanRijnhart committed:
[UPG] 11.0, base_technical_features

osoul-dev committed:
Translated using Weblate (Arabic)

Currently translated at 75.0% (6 of 8 strings)

Translation: server-ux-11.0/server-ux-11.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-11-0/server-ux-11-0-base_technical_features/ar/

hhgabelgaard committed:
Translated using Weblate (Danish)

Currently translated at 75.0% (6 of 8 strings)

Translation: server-ux-11.0/server-ux-11.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-11-0/server-ux-11-0-base_technical_features/da/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-11.0/server-ux-11.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-11-0/server-ux-11-0-base_technical_features/es/

Commit:
Translated using Weblate (Catalan)

Currently translated at 100,0% (8 of 8 strings)

Translation: server-ux-11.0/server-ux-11.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-11-0/server-ux-11-0-base_technical_features/ca/

Commit:
[12.0-mig-base] [MIG] base_technical_features: Migration to 12.0

primes2h committed:
[FIX] travis error: duplicate implicit target name

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-12.0/server-ux-12.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_technical_features/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-12.0/server-ux-12.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_technical_features/it/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-12.0/server-ux-12.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_technical_features/pt/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-12.0/server-ux-12.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_technical_features/zh_CN/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-12.0/server-ux-12.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_technical_features/pt_BR/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 87.5% (7 of 8 strings)

Translation: server-ux-12.0/server-ux-12.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_technical_features/hr/

AaronHForgeFlow committed:
[IMP]base_tchnical_features. Assign automatically to admin not to root

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-12.0/server-ux-12.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_technical_features/hr/

jimhoefnagels committed:
[IMP] base_technical_features: black, isort

jimhoefnagels committed:
[MIG] base_technical_features: Migration to 13.0

sbidoul committed:
pre-commit update

yostashiro committed:
Added translation using Weblate (Japanese)

Commit:
[MIG] base_technical_features: Migration to 14.0

Commit:
Fix linting

Commit:
Fix linting

StefanRijnhart committed:
[FIX] base_technical_features: user setting in hidden group

ThomasBinsfeld committed:
[IMP] base_technical_features: use has_group instead of loading all users
On databases with a large number of users, loading all users of base.group_no_one can be slow.
Moreover, the 'technical_features' and 'show_technical_features' are always computed for a single record

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-14.0/server-ux-14.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-14-0/server-ux-14-0-base_technical_features/es/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-14.0/server-ux-14.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-14-0/server-ux-14-0-base_technical_features/nl/

JuMiSanAr committed:
[IMP] base_technical_features: black, isort, prettier

JuMiSanAr committed:
[MIG] base_technical_features: Migration to 15.0

apineux committed:
[IMP] base_technical_features: black, isort, prettier

apineux committed:
[MIG] base_technical_features: Migration to 16.0

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-16.0/server-ux-16.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-base_technical_features/es/

moitabenfdz committed:
[IMP] base_technical_features: pre-commit stuff

moitabenfdz committed:
[MIG] base_technical_features: Migration to 17.0

Commit:
[UPD] Update base_technical_features.pot

OCA-git-bot committed:
[BOT] post-merge updates

ramiadavid committed:
[IMP] date_range: pre-commit auto fixes

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: server-ux-17.0/server-ux-17.0-base_technical_features
Translate-URL: https://translation.odoo-community.org/projects/server-ux-17-0/server-ux-17-0-base_technical_features/it/

bealdav committed:
[ADD][16.0] developer_menu module

mymage committed:
Added translation using Weblate (Italian)
Currently translated at 100.0% (15 of 15 strings)

Translation: server-ux-16.0/server-ux-16.0-developer_menu
Translate-URL: https://translation.odoo-community.org/projects/server-ux-16-0/server-ux-16-0-developer_menu/it/

Kimkhoi3010 committed:
[MIG] base_technical_features: Migration to 18.0

nilshamerlinck commented on base_technical_features/models/base.py:
```diff
@@ -0,0 +1,18 @@
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def user_has_groups(self, groups):
```
`user_has_groups` has been removed in 18: see `https://github.com/odoo/odoo/pull/151597`

nilshamerlinck commented on base_technical_features/models/ir_ui_view.py:
```diff
@@ -0,0 +1,90 @@
+    _inherit = "ir.ui.view"
+
+    @api.model
+    def _postprocess_access_rights(self, tree):
```
an explanation of the new approach in the PR description would be welcome to help reviewer

nilshamerlinck replied:
still no explanation

nguyenminhchien commented on base_technical_features/models/base.py:
```diff
@@ -6,7 +6,7 @@
     _inherit = "base"
 
-    def user_has_groups(self, groups):
+    def has_groups(self, groups):
```
this method should be in res.users and consider to override `def has_group()` or `def _has_group()`

nguyenminhchien replied:
this commit made me confused, these changes should be included in a migration commit

nguyenminhchien commented on base_technical_features/models/ir_ui_view.py:
```diff
@@ -0,0 +1,90 @@
+    _inherit = "ir.ui.view"
+
+    @api.model
+    def _postprocess_access_rights(self, tree):
```
Overriding this method conflicts with the module `web_studio`

nguyenminhchien replied:
@Kimkhoi3010 
If you cannot find a better way. try this one 
```
    @api.model
    def _postprocess_access_rights(self, tree):
        arch = etree.tostring(tree, encoding="unicode").replace('\t', '').replace('base.group_no_one', 'base_technical_features.group_technical_features')
        tree = etree.fromstring(arch)
        return super()._postprocess_access_rights(tree)
        
```

OCA-git-bot committed:
Initial commit

bealdav committed:
[IMP] developer_menu: apply pre-commit auto fixes

bealdav committed:
[MIG] developer_menu: Migration to 18.0

nguyenminhchien commented on base_technical_features/models/ir_ui_view.py:
```diff
@@ -0,0 +1,36 @@
+            .replace(
+                str(self.env.ref("base.group_no_one").id),
+                str(
+                    self.env.ref("base_technical_features.group_technical_features").id
```
@Kimkhoi3010 replace the string `base.group_no_one` by the string `base_technical_features.group_technical_features`.

NOT replace its database `id`

nguyenminhchien commented on base_technical_features/readme/CREDITS.md:
```diff
@@ -0,0 +1,3 @@
+The migration of this module from 17.0 to 18.0 was financially supported by:
+
+- Camptocamp.
```
Just keep it as before: `The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp.`

nguyenminhchien commented on base_technical_features/tests/test_base_technical_features.py:
```diff
@@ -32,8 +32,11 @@ def get_partner_field():
+        print("\n====", self.env.user.groups_id)
         self.assertEqual(len(get_partner_field()), 0)
         self.env.user.write({"technical_features": True})
+        print("\n====", self.env.user.groups_id)
```
remove 2 print commands

nguyenminhchien commented:
seems that you cannot port this commit in https://github.com/OCA/server-ux/pull/411
so, add it to port blacklist by creating a new commit, include that new commit into this PR

bealdav committed:
add checklog-odoo.cfg

OCA-git-bot committed:
Merge PR #950 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update developer_menu.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-18.0/server-ux-18.0-developer_menu
Translate-URL: https://translation.odoo-community.org/projects/server-ux-18-0/server-ux-18-0-developer_menu/

Kimkhoi3010 committed:
oca-port: blacklist PR(s) 411 for base_technical_features

Kimkhoi3010 commented:
I created a new PR [OCA#953](https://github.com/OCA/server-ux/pull/953) to OCA/server-ux.
(This PR was closed auto after rebased)
